### PR TITLE
Feature/guard for student teacher

### PR DESCRIPTION
**Pull request by trotsm**
Updated: Changed logic of guard works. Only guards of users, who have access are inserted in routes.

Created guards (CanActivate and CanLoad) for students and teachers:
- students have access to shell and student-book;
- teachers have access to shell, progress and journal.

### CODE REVIEW

File: src/app/shell/shell-routing.module.ts [hunk 1]
```diff
@@ -2,28 +2,32 @@ import { NgModule } from '@angular/core';
...
-    canLoad: [AdminPanelGuard],
+    canActivate: [AdminPanelGuard, StudentGuard],
+    canLoad: [AdminPanelGuard, StudentGuard],
     loadChildren: '../journal/journal.module#JournalModule'
   },
   {
     path: 'progress',
-    canActivate: [AdminPanelGuard],
-    canLoad: [AdminPanelGuard],
+    canActivate: [AdminPanelGuard, StudentGuard],
```
Comment (dmitryy90) [inline src/app/shell/shell-routing.module.ts]: progress is for teachers

Comment (trotsm) [reply src/app/shell/shell-routing.module.ts]: Fixed.

File: src/app/services/admin-panel.guard.ts [hunk 2]
```diff
@@ -12,31 +12,31 @@ export class AdminPanelGuard implements CanLoad, CanActivate {
...
    * checks permission every time
-   * @returns true if user's role is not ROLE_ADMIN
-   * @returns false if user's role is ROLE_ADMIN
+   * @returns true if user's role is ROLE_ADMIN
+   * @returns false if user's role is not ROLE_ADMIN
    */
   canActivate(
     route: ActivatedRouteSnapshot,
     state: RouterStateSnapshot): boolean  {
     if (this.isAdmin()) {
```
Comment (dmitryy90) [inline src/app/services/admin-panel.guard.ts]: it could be simplified. return isAdmin()

File: src/app/services/admin-panel.guard.ts [hunk 3]
```diff
@@ -12,31 +12,31 @@ export class AdminPanelGuard implements CanLoad, CanActivate {
...
+   * Function that allows user to navigate to current route modules if it returns true
    * checks permissions only first time
-   * @returns true if user's role is not ROLE_ADMIN
-   * @returns false if user's role is ROLE_ADMIN
+   * @returns true if user's role is ROLE_ADMIN
+   * @returns false if user's role is not ROLE_ADMIN
    */
   canLoad(): boolean {
     if (this.isAdmin()) {
-      return false;
```
Comment (dmitryy90) [inline src/app/services/admin-panel.guard.ts]: same

File: src/app/services/admin-panel.guard.ts (hunk 2 above)
Comment (trotsm) [reply src/app/services/admin-panel.guard.ts]: fixed

**Pull request merged**